classes/book_label.rb

```diff
@@ -0,0 +1,57 @@
+# This is book-label.rb it reps an item class
+module BookLabel
+  def add_book
+    publish_date = get_user_input('Enter the date of publication (yyyy-mm-dd):')
+    publisher = get_user_input('Enter the publisher\'s name:')
+    cover_state = user_cover_state
+    title = get_user_input('Enter the label title:')
+    color = get_user_input('Enter the color of the book')
+
+    book = create_book(publish_date, publisher, cover_state)
+    label = create_label(title, color)
+
+    @books.push(book)
+    @labels.push(label)
+
+    puts 'You\'ve successfully added a book'
+  end
+
+  def get_user_input(prompt)
+    puts prompt
+    gets.chomp
+  end
+
+  def user_cover_state
+    loop do
+      puts 'Enter the state of the book cover (Good or bad):'
+      cover_state = gets.chomp.downcase
+      return cover_state if %w[good bad].include?(cover_state)
+
+      puts 'Invalid input. Please enter "Good" or "Bad".'
+    end
+  end
+
+  def create_book(publish_date, publisher, cover_state)
+    Book.new(publish_date, publisher, cover_state)
+  end
+
+  def create_label(title, color)
+    Label.new(title, color)
+  end
+
+  def list_all_books
+    puts 'Books unavailable' if @books.empty?
+    puts
+    @books.each do |book|
+      puts "Publish Date: #{book.publish_date}, Publisher: #{book.publisher}, Cover State: #{book.cover_state}"
+    end
+  end
+
+  def list_all_labels
+    puts 'Labels unavailable' if @labels.empty?
+    puts
+    @labels.each do |label|
+      puts "Title: #{label.title}, Author: #{label.color}"
+    end
```

Comment:
- [ ] Great effort working on the project. But I would like to raise some issues in order to make your project look perfect and for a better user experience.
- In the label, after adding all the necessary data in there and trying to list the label, I realized that the author was assigned to where the color was supposed to be assigned. I would kindly suggest you change it from `author` to `color`

Screenshot 👇 

![catalog1](https://github.com/2004-okari/CATALOG/assets/78634841/952059e5-219c-4ec5-8d9c-9ec6a6d817bf)